### ISSUE TITLE
Location services requirement for photo instant upload

### ISSUE THREAD

**Issue by gbiggs** [owncloud/ios#370]
Enabling the instant upload feature requires that the Owncloud client have access to location services. This in turn causes location services to be constantly switched on as the Owncloud client is running in the background.

Location services should not be a requirement for instant upload of photos. It should be possible to forbid the Owncloud client from accessing location services and still upload photos taken instantly.

Either:
- The Owncloud client could automatically detect that no location information is available and skip doing whatever it is doing with the location information before uploading the photo, or
- There could be a setting to choose to do whatever is done with the location information before uploading, and enabling that setting requires permitting access to location services.

**Comment by javiergonzper** [comment]
@gbiggs the ownCloud iOS App use the location to wake up the app in background in order to check if any there is any new image to be uploaded.

Maybe we have to add more options because maybe some users just want to check if there is new images when they open the ownCloud app.

**Comment by gbiggs** [comment]
What is the location event that is being used to wake up the app in the background? Is it just being used as a timer event? (Does this mean that iOS doesn't have any API to periodically wake up an app that is in the background to perform some processing?)

Certainly giving up automatic uploads when the app is in the background is an option for users who do not want to have location services switched on, but perhaps we can find a better solution.

**Comment by javiergonzper** [comment]
The app wake up when the user make "significant change location". On that moment we check if there are any pictures in order to upload it.
Any new idea is welcome :+1:

**Comment by marinotws** [comment]
Unfortunately, I think there are no other ways ... :(

**Comment by gbiggs** [comment]
I looked at the iOS developer's documentation, and indeed it looks like there is no other option for background execution of photo uploading. Until Apple implements and API hook for when a photo is taken, using the location services to wake the app is probably the best that can be done.

I think that the usability can be improved, though. There should be two options in the settings. The first would be to upload photos automatically; enabling it would cause new photos to be uploaded automatically when the Owncloud app is opened. This would not require location services. A second option would enable automatic uploading even when the app is not open, but would require location services to be enabled. Why location services is required should be made clear to the user.

If Apple ever implements an API hook for new photos, the options would not need to be changed. All that would change would be removing the use of location services for the background upload option.

**Comment by OC2015** [comment]
OK I have understood that the "behavior" is iOS related

It would be great to have an "Upload now" button within the app to upload all new images while there is no other resolution possible.

Thank you in advance and best regards
OC2015

**Comment by stonie08** [comment]
I agree an "Upload now" Button would be really nice. Either implemented in the app and/or as a widget for the notification area that says "n images available for upload". Although I don't know if the second one is possible.

**Comment by privacyclown** [comment]
Hi there,

I also think this is something that needs to be fixed.  I don't use location services except when I have to.  I have GPS off most of the time and enable it when I need to use it for Uber/Maps or whatever, I don't like companies tracking my movements.

First problem is the app is set to use location "Always" this needs to be changed to "When in Use" - An upload now button is the quck fix but I'd much prefer you use a timer, say once an hour to wake the app and upload any new images it finds.

That all said thanks for giving me back my privacy, you guys do great work :)

**Comment by nasli** [comment]
Hi @privacyclown, 
your privacy is assured, we don't track any of your movements. We just use an event that provide the device when it detects a big change in location. 

We need to use the location to have and event that wake up the app in order to check in that moment whether there are new photos available to keep them save as soon as possible but without draining the battery, is for that that we only use the significant changes of location event and not gps. 

No doubt that this feature should improve and add new settings options for any user can customize to their needs. Will be added in future versions, and also contributions available :)

**Comment by privacyclown** [comment]
Hi @nasli I totally trust that you don't - but as a personal choice I don't enable location services so companies like google and apple who don't neccessarily share your values can track me, it comes back to I want control of my device and the information that can be logged about my movements via GPS.

**Comment by privacyclown** [comment]
also - keep up the great work! :)

**Comment by stil** [comment]
Yesterday I installed ownCloud iOS app and I also noticed this strange behaviour. Many people will get suspicious if you won't add some explaination why is it needed. I'm not an iOS developer, but since version 7.0 we have something called "background app refresh". Isn't it possible to make it uploading new photos to server?

**Comment by javiergonzper** [comment]
No @stil that functionality need a background notification from the server to wake up the mobile app in order to check if there is anything new on the server to update the information.

If we only have a server we could develop a background notification system but with ownCloud every user could have his own server so... it is not possible to use that feature.

**Comment by cdamken** [comment]
@javiergonzper @rperezb Will ownCloud apps (iOS, Android, Desktop) attempt to exchange data with servers that we did not specify in your ownBrander application?

**Comment by rperezb** [comment]
@cdamken I don´t fully understand the use case, can you please provide further details?

**Comment by cdamken** [comment]
@rperezb Does the app exchange data with some servers when they try to get the location? if so, with which servers is the app connected? and does the server store some information from the app?

**Comment by nasli** [comment]
@cdamken, the iOS system provided us when a signicant change in location was happened, no servers involved, no exchange data. 

The significant-change location service allow us to wake up the iOS app that is currently suspended or not running in order to check if there are new photos.

**Comment by cdamken** [comment]
@nasli Thanks a lot for the info!

**Comment by aarondunlap** [comment]
Have there been any significant changes in iOS 9 that would allow this feature to be reimplemented in a more elegant way?

**Comment by rperezb** [comment]
@aarondunlap no, for the moment, we have not return to this feature.
FYI, We are focus on the internal sharing and download folder, as well as fixing some bugs.

**Comment by nasli** [comment]
@aarondunlap, not yet :(
Some updates about that here #511

**Comment by binou06** [comment]
Hello, for me, not an issue to upload photo only when I open the application. (this is how it works for dropbox no ? It only upload photos when launching dropbox application).

Could be smart to keep application like it but don't disable uploading when disabling location... In this case, only syncing when launching application.

That being said, great work !

Cheers

**Comment by Relequestual** [comment]
Where do I find information on when the 3.5.0 release will be complete?

Thanks in advance. Great to see this issue has been resolved.

**Comment by michaelstingl** [comment]
@Relequestual see https://github.com/owncloud/ios/milestones/3.5.0

**Comment by Relequestual** [comment]
Thanks @michaelstingl, I did see that, but hard to tell with projects how closely they stick to using them as indicators for release. Looks like it's pretty close then! Impressed with the level of organisation in this project! =]

**Comment by nasli** [comment]
@Relequestual we expect in the next weeks ;)

**Comment by Relequestual** [comment]
Thanks @nasli for updating me. I was thinking about trying to help out here, but it's unclear what all the labels are for. Shall I create an issue for this or is it somewhere else more central that I've missed?

**Comment by rperezb** [comment]
@Relequestual we also encourage you to install zenhub so that you may have an overview of the iOS app status and priorization thanks to the board. We are using the zenhub status to track the priorities 
Certainly, you may help us, very appreciated!!

**Comment by Relequestual** [comment]
Great, thanks for this! I'll look into that. Looks pretty neat! =]

**Comment by Relequestual** [comment]
Hum.
Documentation still says that location must be enabled:

https://doc.owncloud.com/ios/#instant-upload-of-photos-and-videos

Is this right or wrong?
